Sources/Atomic.swift

```diff
@@ -104,14 +176,13 @@ internal final class RecursiveAtomic<Value>: AtomicProtocol {
 	///
 	/// - returns: The result of the action.
 	@discardableResult
+	@inline(__always)
 	func modify<Result>(_ action: (inout Value) throws -> Result) rethrows -> Result {
 		lock.lock()
-		defer {
-			didSetObserver?(_value)
-			lock.unlock()
-		}
-
-		return try action(&_value)
+		let returnValue = try action(&_value)
+		didSetObserver?(_value)
+		lock.unlock()
```

Comment:
Without `defer`, if `try action(&_value)` throws, the lock will never be unlocked.